lib/solar_system.rb

```diff
@@ -0,0 +1,76 @@
+# lib/solar_system.rb
+
+class SolarSystem
+
+  attr_reader :star_name, :planets, :planets_by_name
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+    #@planets_by_name = []
+  end
+
+  def add_planet(planet)
+    # don't do this
+    raise ArgumentError, "Argument must be instance of Planet class" unless planet.is_a? Planet
+    @planets << planet
+  end
+
+def add_planet_name(planet)
+  @planets_by_name = @planets.map{ |p| p.name.upcase }
+end
+
+  def list_planets
+    list = "Planets orbiting #{@star_name}:\n"
+    @planets.each_with_index {|p, i| list << "#{i+1}. #{p.name}\n" }
+    return list
```

Comment:
This is returning an array of strings, not a string.

Reply:
I think it's a string!